dbms/src/Functions/FunctionsTiDBConversion.h

```diff
@@ -1610,6 +1611,47 @@ struct TiDBConvertToDuration
                 block.getByPosition(result).column = std::move(to_col);
             }
         }
+        else if constexpr (std::is_same_v<FromDataType, DataTypeMyDate>)
+        {
+            // cast date as duration
+            auto to_col = ColumnVector<Int64>::create();
+            auto & vec_to = to_col->getData();
+            vec_to.resize(size);
+            for (size_t i = 0; i < size; ++i)
+            {
+                vec_to[i] = 0; // The DATE type is used for values with a date part but no time part. The value of Duration is always zero.
+            }
+            block.getByPosition(result).column = std::move(to_col);
```

Comment:
Since the result column is always 0, can we use ColumnConst to optimize a little bit?

Reply:
thx,done

Reply:
thx,done

Reply:
`block.getByPosition(result).column = std::make_shared<DataTypeInt64>()->createColumnConst(size, 0);`